From: Thomas Koch <linrunner@gmx.net>
Date: Wed, 15 Jan 2014 20:42:05 +0100
Subject: Fix: "Use of uninitialized value $aslBase in concatenation (.) or
 string at tpacpi-bat line 401."

Move the verbose print statement within the scope of the $aslBase loop.
---
 tpacpi-bat | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/tpacpi-bat b/tpacpi-bat
index 205c2f2..6d84123 100755
--- a/tpacpi-bat
+++ b/tpacpi-bat
@@ -393,13 +393,15 @@ sub acpiCall($){
     die "Could not find $acpiCallDev. Is module acpi_call loaded?\n";
   }
 
-  my ($aslBase, $val);
+  my ($val);
   foreach my $aslBase(@aslBases) {
     $val = runAcpiCall $call, $aslBase;
-    last if not $val =~ /Error: AE_NOT_FOUND/;
+    if (not $val =~ /Error: AE_NOT_FOUND/){
+      print "Call    : $aslBase.$call\n" if $verbose;
+      print "Response: $val\n" if $verbose;
+      last
+    }
   }
-  print "Call    : $aslBase.$call\n" if $verbose;
-  print "Response: $val\n" if $verbose;
   return $val;
 }
 sub runAcpiCall($$){
